parcels/particlefile/baseparticlefile.py

```diff
@@ -92,12 +94,16 @@ def __init__(self, name, particleset, outputdt=np.infty, write_ondelete=False, c
             self.time_origin = self.particleset.time_origin
             self.lonlatdepth_dtype = self.particleset.collection.lonlatdepth_dtype
             self.var_names = []
+            self.var_dtypes = []
             self.var_names_once = []
+            self.var_dtypes_once = []
             for v in self.particleset.collection.ptype.variables:
-                if v.to_write == 'once':
+                if v.to_write == 'once' and self.write_ondelete is False:
                     self.var_names_once += [v.name]
-                elif v.to_write is True:
+                    self.var_dtypes_once += [v.dtype.__name__]
```

Comment:
Shy not store the `dtype` object, instead of its name?
```suggestion
                    self.var_dtypes_once += [v.dtype]
```